src/lsp/cobol_parser/text_tokenizer.ml

```diff
@@ -153,6 +154,183 @@ let preproc_n_combine_tokens ~source_format =
   let rec skip ((p', l', dgs) as acc) ((p, l) as pl) = function
     | 0 -> acc, pl
     | i -> skip (hd p :: p', hd l :: l', dgs) (tl p, tl l) (i - 1)
+  and subst_function ?(in_repo = false) ((p', l', dgs) as acc) ((p, l) as pl) =
```

Comment:
Surely some more help from the lexer will help keep that clean.

Reply:
This part should be removed or much simplified: we need to avoid combining `FUNCTION` tokens if at all possible.